### PR TITLE
[#2221] Private type aliases

### LINKED ISSUE #2221: Make all type hint aliases private
Types like `HTTPBody`, `PeerCertRetType`, etc should be private so users don't think they're supported features of our API.

### PR DESCRIPTION
**Pull request by csurfer**
This PR converts the following type aliases to private.

1. `HTTPBody` to `_TYPE_HTTP_BODY`
2. `ReduceResult` to `_TYPE_REDUCE_RESULT`
3. `SocketOptions` to `_TYPE_SOCKET_OPTIONS`
4. `PeerCertRetDictType` to `_TYPE_PEER_CERT_RET_DICT`
5. `PeerCertRetType` to `_TYPE_PEER_CERT_RET`

### Notes

1. Type definitions within `contrib/_securetransport/bindings.py` have been ignored in this PR.
2. `nox --reuse-existing-virtualenvs --sessions format` has been run on every commit.

Closes #2221

### CODE REVIEW

**Comment by csurfer** [comment]
@sethmlarson Please take a look and let me know what other changes you require on this PR.

File: src/urllib3/connection.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ class BaseSSLError(BaseException):  # type: ignore
 _CONTAINS_CONTROL_CHAR_RE = re.compile(r"[^-!#$%&'*+.^_`|~0-9a-zA-Z]")
 
 
-HTTPBody = Union[bytes, IO[Any], Iterable[bytes], str]
+_TYPE_HTTP_BODY = Union[bytes, IO[Any], Iterable[bytes], str]
```
Comment (sethmlarson) [inline src/urllib3/connection.py:77]: For consistency let's name this `_TYPE_BODY`

File: src/urllib3/connection.py [hunk 2]
```diff
@@ -286,7 +288,7 @@ def request_chunked(
         self,
         method: str,
         url: str,
-        body: Union[None, HTTPBody, Tuple[Union[bytes, str]]] = None,
+        body: Union[None, _TYPE_HTTP_BODY, Tuple[Union[bytes, str]]] = None,
```
Comment (sethmlarson) [inline src/urllib3/connection.py:291]: Seeing this here has alerted me to a type issue, woo!

This type here should be `Optional[_TYPE_BODY]` and line 316 should be changed to `body = (to_bytes(body),)`. You'll have to import `to_bytes` above as well.

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Thanks for this! Your PR actually brought a type issue to my attention which is super great. Couple of comments for you:

**Comment by sethmlarson** [comment]
Pinging @hramezani as the resident type-master :)

File: src/urllib3/connection.py (hunk 1 above)
Comment (csurfer) [reply src/urllib3/connection.py:77]: Done.

File: src/urllib3/connection.py (hunk 2 above)
Comment (csurfer) [reply src/urllib3/connection.py:291]: Done.

File: src/urllib3/connection.py [hunk 3]
```diff
@@ -286,7 +288,7 @@ def request_chunked(
         self,
         method: str,
         url: str,
-        body: Union[None, HTTPBody, Tuple[Union[bytes, str]]] = None,
+        body: Optional[_TYPE_BODY] = None,
```
Comment (sethmlarson) [inline src/urllib3/connection.py:291]: :tada:

**Review by sethmlarson — APPROVED** [review]
LGTM, will merge after @hramezani gets to weigh in too.

**Review by hramezani — APPROVED** [review]
LGTM! Nice work :+1:

**Comment by sethmlarson** [comment]
Thanks @csurfer! :clap:

**Pull request merged**